allenact/algorithms/onpolicy_sync/engine.py

```diff
@@ -651,6 +658,18 @@ def __init__(
 
         self.training_pipeline: TrainingPipeline = config.training_pipeline()
 
+        if self.num_workers != 1:
+            # Ensure that we're only using early stopping criterions in the non-distributed setting.
+            if any(
+                stage.early_stopping_criterion is not None
+                for stage in self.training_pipeline.pipeline_stages
+            ):
+                raise NotImplementedError(
+                    "Early stopping criterions are currently only allowed when using a single training worker, i.e."
+                    " no distributed (multi-GPU) training. If this is a feature you'd like please create an issue"
+                    " at https://github.com/allenai/allenact/issues."
```

Comment:
I would add "... or feel free to make a PR with your own implementation for us to review".